providers/os/resources/purl/platform_purl_test.go

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) Mondoo, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package purl
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.mondoo.com/cnquery/v11/providers-sdk/v1/inventory"
+)
+
+func TestNewPlatformPurl(t *testing.T) {
+	tests := []struct {
+		name     string
+		platform *inventory.Platform
+		want     string
+		wantErr  string
+	}{
+		{
+			name: "valid ubuntu platform",
+			platform: &inventory.Platform{
+				Name:    "ubuntu",
+				Version: "22.04",
+			},
+			want:    "pkg:os/ubuntu/@22.04?distro=ubuntu-22.04",
+			wantErr: "",
+		},
+		{
+			name: "valid windows platform",
+			platform: &inventory.Platform{
+				Name:    "windows",
+				Version: "19045",
+			},
+			want:    "pkg:os/windows/@19045?distro=windows-19045",
+			wantErr: "",
+		},
+		{
+			name: "platform with arch",
+			platform: &inventory.Platform{
+				Name:    "ubuntu",
+				Version: "22.04",
+				Arch:    "amd64",
+			},
+			want:    "pkg:os/ubuntu/@22.04?arch=amd64&distro=ubuntu-22.04",
+			wantErr: "",
+		},
+		{
+			name: "platform with x86_64 arch",
+			platform: &inventory.Platform{
+				Name:    "ubuntu",
+				Version: "22.04",
+				Arch:    "x86_64",
+			},
+			want:    "pkg:os/ubuntu/@22.04?arch=x86_64&distro=ubuntu-22.04",
+			wantErr: "",
+		},
+		{
+			name: "platform with arm64 arch",
+			platform: &inventory.Platform{
+				Name:    "ubuntu",
+				Version: "22.04",
+				Arch:    "arm64",
+			},
+			want:    "pkg:os/ubuntu/@22.04?arch=arm64&distro=ubuntu-22.04",
+			wantErr: "",
+		},
+		{
+			name: "macos with apple silicon",
+			platform: &inventory.Platform{
+				Name:    "macos",
+				Version: "14.5.1",
+				Arch:    "arm64",
+			},
+			want:    "pkg:os/macos/@14.5.1?arch=arm64&distro=macos-14.5.1",
+			wantErr: "",
+		},
+		{
+			name: "windows with x86 arch",
+			platform: &inventory.Platform{
+				Name:    "windows",
+				Version: "19045",
+				Arch:    "x86",
+			},
+			want:    "pkg:os/windows/@19045?arch=x86&distro=windows-19045",
+			wantErr: "",
+		},
+		{
+			name: "vsphere platform",
+			platform: &inventory.Platform{
+				Name:    "vsphere",
+				Version: "7.0.3",
+			},
+			want:    "pkg:os/vsphere/@7.0.3?distro=vsphere-7.0.3",
+			wantErr: "",
+		},
+		{
+			name: "esxi platform",
+			platform: &inventory.Platform{
+				Name:    "esxi",
+				Version: "7.0.3",
+				Arch:    "x86_64",
+			},
+			want:    "pkg:os/esxi/@7.0.3?arch=x86_64&distro=esxi-7.0.3",
+			wantErr: "",
+		},
+		{
+			name: "kubernetes deployment",
+			platform: &inventory.Platform{
+				Name:    "k8s-deployment",
+				Version: "1.27",
+			},
+			want:    "pkg:os/k8s-deployment/@1.27?distro=k8s-deployment-1.27",
+			wantErr: "",
+		},
+		{
+			name: "aws platform",
+			platform: &inventory.Platform{
+				Name: "aws",
+			},
+			want:    "pkg:os/aws/?distro=aws",
```

Comment:
Same for cloud providers.

Other terms I could think of:
- platform
- runtime

???

Reply:
Let's switch to platform